### PR TITLE
Battle agent

### PR DESCRIPTION
**Pull request by thyo9470**
- Added a battle agent which can make decisions about what move the enemy should take
    - The agent uses minimax search and you give it the depth you want the enemy to search to

- Added smart walls and new sprites

### CODE REVIEW

File: Game/sources/BattleSim.cpp [hunk 1]
```diff
@@ -2,27 +2,33 @@
...
 
   // make enemy
   QJsonObject entity_data;
   int enemy_level =  player->GetLevel() + ( qrand() % 5 - 2 );
+  //int enemy_level = 10;
   enemy_level = (enemy_level<1)?1:enemy_level;
   entity_data["level"] = enemy_level;
-  entity_data["max_health"] = enemy_level * 10;
-  entity_data["max_magic"] = enemy_level * 10;
+  entity_data["max_health"] = floor(log(enemy_level+1)/log(1.017)); // FIX FLOATING NUMBERS
```
Comment (barribob) [inline Game/sources/BattleSim.cpp:22]: I forgot what this was doing exactly.

File: Game/sources/FightWindow.cpp [hunk 2]
```diff
@@ -82,19 +112,23 @@ void FightWindow::UpdateFightWindow(BattleSim* battle_sim){
...
-  ui->enemyMagicBar->setValue(enemy->GetMagicPercent());
+  ui->enemyHealthBar->setMaximum(enemy->GetMaxHealth());
+  ui->enemyHealthBar->setValue(enemy->GetHealth());
+  ui->enemyMagicBar->setMaximum(enemy->GetMaxMagic());
+  ui->enemyMagicBar->setValue(enemy->GetMagic());
 
   // Create and add the tile to the scene
   QGraphicsPixmapItem * pixmap_enemy = new QGraphicsPixmapItem();
-  pixmap_enemy->setPixmap(sprite_sheet_.copy(enemy->get_sprite_index() * sprite_size_, row * sprite_size_, sprite_size_, sprite_size_));
+  //pixmap_enemy->setPixmap(sprite_sheet_.copy(enemy->get_sprite_index() * sprite_size_, row * sprite_size_, sprite_size_, sprite_size_));
```
Comment (barribob) [inline Game/sources/FightWindow.cpp]: Maybe remove this comment?

File: Game/sources/Window.cpp [hunk 3]
```diff
@@ -50,26 +54,226 @@ void Window::keyPressEvent(QKeyEvent *event)
...
+
+        pixmap = GetWallSprite(walls);
+
+      }else{
+        pixmap = GetDungeonSprite(dungeon_tile);
+
+      }
+
+     // pixmap = GetDungeonSprite(dungeon_tile);
+      //pixmap->setPixmap(dungeon_sheet_.copy(tile_pos_x * dungeon_sprite_size_, tile_pos_y * dungeon_sprite_size_, dungeon_sprite_size_, dungeon_sprite_size_));
```
Comment (barribob) [inline Game/sources/Window.cpp]: Do we need these comments?

**Review by barribob — APPROVED** [review]
Looks good! Only thing I noticed is that the loading and saving the board doesn't load in the void tile. You  can probably go into the Board::Read function and add a new case statement so that it also loads in the void tiles.

File: Game/sources/BattleSim.cpp (hunk 1 above)
Comment (thyo9470) [reply Game/sources/BattleSim.cpp:22]: This sets their max health depending on their level. I am going to make it not magic numbers but for not it's just hard coded. I can also fix the void issue.

**Pull request merged**